### PR TITLE
feat: remove 'v' and add release date for changelog generation

### PR DESCRIPTION
**Pull request by tadelesh**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by tadelesh** [comment]
Releated PR: https://github.com/Azure/azure-sdk-for-go/pull/15524

**Review by weshaggard — APPROVED** [review]

**Comment by RickWinter** [comment]
@weshaggard is Central-EngSys planning to consolidate this logic with the data plane logic?

**Comment by weshaggard** [comment]
@RickWinter There aren't plans from Central-EngSys to do this. However, I do think mgmt and data plane teams should work together to figure out the correct set of tools/scripts to use for the codegen automation and EngSys can assist in guidance and reviews as necessary.

File: tools/generator/cmd/v2/common/fileProcessor.go [hunk 1]
```diff
@@ -228,7 +229,10 @@ func AddChangelogToFile(changelog *model.Changelog, version *semver.Version, pac
 	oldChangelog := string(b)
 	insertPos := strings.Index(oldChangelog, "##")
 	additionalChangelog := changelog.ToCompactMarkdown()
-	newChangelog := oldChangelog[:insertPos] + "## v" + version.String() + " (released)\n" + additionalChangelog + "\n\n" + oldChangelog[insertPos:]
+	if releaseDate == "" {
+		releaseDate = time.Now().Format("2006-01-02")
```
Comment (benbp) [inline tools/generator/cmd/v2/common/fileProcessor.go:233]:
```suggestion
		releaseDate = time.Now().UTC().Format("2006-01-02")
```

File: tools/generator/cmd/template/templateCmd.go [hunk 2]
```diff
@@ -119,13 +123,18 @@ func GeneratePackageByTemplate(rpName, packageName string, flags Flags) error {
...
-func buildReplaceMap(rpName, packageName, packageTitle, commitID string) {
+func buildReplaceMap(rpName, packageName, packageTitle, commitID, releaseDate string) {
 	replaceMap = make(map[string]string)
 
 	replaceMap[RPNameKey] = rpName
 	replaceMap[PackageNameKey] = packageName
 	replaceMap[PackageTitleKey] = packageTitle
 	replaceMap[CommitIDKey] = commitID
+	if releaseDate == "" {
+		replaceMap[ReleaseDate] = time.Now().Format("2006-01-02")
```
Comment (benbp) [inline tools/generator/cmd/template/templateCmd.go:134]:
```suggestion
		replaceMap[ReleaseDate] = time.Now().UTC().Format("2006-01-02")
```

**Review by benbp — APPROVED** [review]

**Pull request merged**